shared/src/main/scala/pl/metastack/metatime/Offset.scala

```diff
@@ -15,6 +17,34 @@ object Formatter {
   def branch(ranges: Seq[(Range, Any)], fallback: Any): Formatter = ???
 }
 
-trait Offset[T] {
-  def format: String = ???
+trait Offset[+T] {
```

Comment:
Can you add a type constraint (`trait Offset[+T <: Component]`)? Then the cast in line 23 should become redundant.